### PR TITLE
ControlGroup is in charge of its data

### PR DESCRIPTION
**Pull request by erendrake**
if the control group has more control over its own inputs we can get rid of some redundant code.

### CODE REVIEW

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs [hunk 1]
```diff
@@ -658,23 +601,22 @@ private void ControlWindow(int windowID)
                     if (last.Contains(pos) && Event.current.type == EventType.Repaint)
                         tooltipText = "Speed Multiplier";
 
-                    float speed;
-                    bool speedOk = float.TryParse(g.Speed, out speed);
+                    //float speed;
```
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: I am not thinking this is required anymore, but i wasnt 100% sure.

Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: Thing is that keypresses are handled by each servo from inside MuMechToggle, so this part is necessary. And speed parsing is necessary too.

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs [hunk 2]
```diff
@@ -1698,23 +1636,79 @@ public ControlGroup()
...
+                }
+            }
+
+            public string Speed
+            {
+                get { return speed; }
+                set
+                {
+                    speed = value;
+                    PropogateSpeed();
```
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: @ZiwKerman When you set speed it parses it and gives it to all of its servos. that is why i commented the parsing out from above.

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs (hunk 1 above)
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: They are but i dont see why they are needed here. the key is already set when you either add the servo to a group or change the group's key. there didnt seem to be any state change that would require that to get refreshed in this method. 

Also speed parsing is still being done, by the group object

**Pull request merged**